src/main/resources/hudson/tasks/junit/pipeline/JUnitResultsStep/help-ignoreEnclosingDepth.html

```diff
@@ -0,0 +1,36 @@
+<div>
+    Optionally, specify the number of enclosing stages or branches, starting from the innermost stage or branch, to
+    ignore. Defaults to 0, meaning all are included.
+
+    Example:
+    <code>
+        stage('outer stage') {
+          stage('inner stage') {
+            parallel(branchA: {
+                       node {
+                         sh './run_first_set_of_tests.sh`
+                         junit testResults: '**/target/surefire-reports/TEST-*.xml'
+                       }
+                     },
+                     branchB: {
+                       node {
+                         sh './run_second_set_of_tests.sh`
+                         junit testResults: '**/target/surefire-reports/TEST-*.xml', ignoreEnclosingDepth: 1
+                       }
+                     },
+                     branchC: {
+                       node {
+                         sh './run_third_set_of_tests.sh`
+                         junit testResults: '**/target/surefire-reports/TEST-*.xml', ignoreEnclosingDepth: 2
```

Comment:
This seems rather baroque.

Is it too late to turn off the automatic association of results to enclosing blocks, which in my experience just winds up producing weirdly named test sets, and make the test set name be a simple parameter to the `junit` step as in the original proposal? Then we would not need any tricks like this.

Reply:
It is a bit baroque. I do think the automatic logic is valuable, but it may make sense to only pay attention to `parallel`s if a flag is given (or not at all). @i386, @michaelneale - thoughts?

Reply:
> as in the original proposal

At this point I am not sure if I was referring to https://github.com/jenkinsci/junit-plugin/pull/64#issuecomment-281700612 or something else.